src/modules/trajectory_generation/domain/spline_generation/cubic_spline_generator.hpp

```diff
@@ -2,19 +2,160 @@
+
+            /**
+             * @brief Outputs a set of postures after computing the final spline params
+             * 
+             * @return std::vector<core_datastructures::Posture> 
+             */
+            std::vector<core_datastructures::Posture> generate_splines();
+
+            /**
+             * @brief Computes curvature given params and arc length
+             * 
+             * @param params 
+             * @param s 
+             * @return double Curvature
+             */
+            double calculate_kappa(std::vector<double>& params, double s);
+
+            /**
+             * @brief Computes theta given params and arc length
+             * 
+             * @param params 
+             * @param s 
+             * @return double theta value in radians
+             */
+            double calculate_theta(std::vector<double>& params, double s);
+
+            // double calculate_x();
+            /**
+             * @brief computes x given params and arc length
+             * 
+             * @param params 
+             * @param s 
+             * @return double 
+             */
+            double calculate_x(std::vector<double>& params, double s);
+
+            /**
+             * @brief Computes y given params and arc length
+             * 
+             * @param params 
+             * @param s 
+             * @return double 
+             */
+            double calculate_y(std::vector<double>& params, double s);
+
+            /**
+             * @brief Gets the euclidian distance given two postures
+             * 
+             * @param start 
+             * @param goal 
+             * @return double eucilidian distance
+             */
+            double get_distance(const core_datastructures::Posture& start, const core_datastructures::Posture& goal);
+
+            
+            /**
+             * @brief return new vector | goal - start |
+             * 
+             * @param start 
+             * @param goal 
+             * @return Eigen::Vector3d 
+             */
+            Eigen::Vector3d diff(const core_datastructures::Posture& start, const core_datastructures::Posture& goal);
+            
+            /**
+             * @brief checks if qiven vector is greater than threhold
+             * 
+             * @param v1  
+             * @return true if values within range of threhold
+             * @return false if values greater than the threshold
+             */
+            bool comp(Eigen::Vector3d& v1);
+
         private:
+            double P0;
+            double P1;
+            double P2;
+            double P3;
+            double SG;
+            
+            /**
+             * @brief Spline convergence threshold value
+             * 
+             */
+            Eigen::Vector3d q_thresh;
+
+            /**
+             * @brief Start posture for spline generation
+             * 
+             */
             core_datastructures::Posture start;
```

Comment:
This may not be required since we are always starting from 0,0,0,0. (as mentioned in mcnaughton thesis). In the constructor, we can compute the relative goal and the store it.

I may be wrong but I dont think we are using start anywhere else.